scripts/src/main/resources/scripts/command/java

```diff
@@ -96,10 +92,197 @@ function doCreate() {
+      artifactId+=${1}
+    fi
+    shift
+  done
+
+  while [[ -z "${framework}" ]] || [[ "${framework}" != "spring" ]] && [[ "${framework}" != "quarkus" ]]
+  do
+    if [[ -n ${package} ]]
+    then
+      framework="spring"
+      break
+    fi
+    if [[ -z "${framework}" ]]
+    then
+      doEcho "No framework selected." 
+    fi
+    read -r -p "(Select a framework: 'quarkus' or 'spring'): " framework
+  done
+
+  if [[ "${framework}" = "quarkus" ]]
+  then
+    startQuarkusProject "${dbtype}"
+  elif [[ "${framework}" = "spring" ]]
+  then
+    if [[ -z "${package}" ]] && [[ -z "${mavenargs}"  ]]
+    then 
+      doEcho "No package or other parameter passed, we need at least a package to create a spring application more information https://devonfw.com/website/pages/docs/devonfw-guide_ide.wiki_cli.asciidoc.html#devonfw-guide_ide.wiki_java.asciidoc_create" 
+      read -r -p "Pass a package and optional parameters as shown in the examples: " mavenargs
+    fi
+    doMavenArchetype "${package}" "${mavenargs}" "${dbtype}" "${artifactId}"
+  fi
   exit ${?}
 }
 
+# dbtype
+function startQuarkusProject() {
+  local message='\nPlease select your preferred extensions and set the project group-, artifactId-, and version
+  on the opened site https://code.quarkus.io/ or just press return to use a configured project from our 
+  templates guides with some extensions that we recommend to start with \n' 
+  while [[ ! "$url" =~ ^https://code.quarkus.io/.* ]]
+  do
+    doEchoAttention -e "$message"
+    read -r -p "(Copy the url and paste it here) The url has to start like this:https://code.quarkus.io/... or just press return to use our simple REST API template project" url
+    if [[  -z "${url}" ]]
+    then 
+      url="https://code.quarkus.io/?g=com.devonfw.quarkus&e=resteasy&e=resteasy-jackson&e=hibernate-validator&e=hibernate-orm"
+      break
+    fi
+  done
+
+  local artifact
+  artifact="$(echo "${url}" | grep -o 'a=[^\&]*')"
+  if [[ -z "${artifact}" ]]
+  then 
+    artifact="myapp"
+    url="${url}&a=${artifact}"
+  else
+    artifact=${artifact:2}
+  fi
+
+  local group
+  group="$(echo "${url}" | grep -o 'g=[^\&]*')"
+  if [[ -z "${group}" ]]
+  then 
+    group="com.devonfw.quarkus"
+    url="${url}&g=${group}"
+  else
+    group=${group:2}
+  fi
+
+  doEcho "Project will be created in workspaces/main folder"
+  local database_arr=(postgresql mariadb mysql h2 mssql db2 derby oracle dynamodb mongodb redis)
+  local database_extensions_arr=("e=jdbc-postgresql" "e=jdbc-mariadb" "e=jdbc-mysql" "e=jdbc-h2" "e=jdbc-mssql" "e=jdbc-db2" "e=jdbc-derby" "e=jdbc-oracle" "e=amazon-dynamodb" "e=mongodb-client" "e=redis-client")
+  local database_extension
+  if [[ -n "${1}" ]]
+  then 
+    for ((i=0; i< ${#database_arr[@]}; i++))
+    do 
+      if [[ "${1}" == "${database_arr[${i}]}" ]]
+      then 
+        database_extension="${database_extensions_arr[${i}]}" 
+        url+="&${database_extension}&e=hibernate-orm"
```

Comment:
is there a problem if `e=hibernate-orm` was already present in URL and will now be duplicated?

Reply:
I take a look at this, and it seems not to be a problem.